wheel/python/clvm_rs/casts.py

```diff
@@ -0,0 +1,170 @@
+"""
+Some utilities to cast python types to and from clvm.
+"""
+
+from typing import Callable, List, SupportsBytes, Tuple, Union, cast
+
+from .clvm_storage import CLVMStorage, is_clvm_storage
+
+
+AtomCastableType = Union[
+    bytes,
+    str,
+    int,
+    SupportsBytes,
+]
+
+
+CastableType = Union[
+    AtomCastableType,
+    List["CastableType"],
+    Tuple["CastableType", "CastableType"],
+    CLVMStorage,
+]
+
+
+NULL_BLOB = b""
+
+
+def generate_working_memview_or_bytes():
+    # in python3.7 you can't go 'isinstance(b"", SupportsBytes)'
+
+    # this one works in py38+
+    def memview_or_bytes_py38_or_later(o):
+        return isinstance(o, (memoryview, SupportsBytes))
+
+    # this one works in py37
+    def memview_or_bytes_py37(o):
+        return getattr(o, "__bytes__", None) is not None
+
+    try:
+        memview_or_bytes_py38_or_later(b"")
+        return memview_or_bytes_py38_or_later
+    except TypeError:
+        pass
+    return memview_or_bytes_py37
+
+
+memview_or_bytes = generate_working_memview_or_bytes()
+
+
+def int_from_bytes(blob):
+    """
+    Convert a bytes blob encoded as a clvm int to a python int.
+    """
+    size = len(blob)
+    if size == 0:
+        return 0
+    return int.from_bytes(blob, "big", signed=True)
+
+
+def int_to_bytes(v) -> bytes:
+    """
+    Convert a python int to a blob that encodes as this integer in clvm.
+    """
+    byte_count = (v.bit_length() + 8) >> 3
```

Comment:
is there really a good reason to use `>> 3` instead of `// 8`?

Reply:
I assumed it would be faster, or maybe compile to the same thing. Turns out I was wrong.

```
Disassembly of <code object div8 at 0x100432ef0, file "shift_vs_div.py", line 1>:
  2           0 LOAD_FAST                0 (n)
              2 LOAD_CONST               1 (8)
              4 BINARY_FLOOR_DIVIDE
              6 RETURN_VALUE

Disassembly of <code object shift3 at 0x100432fa0, file "shift_vs_div.py", line 5>:
  6           0 LOAD_FAST                0 (n)
              2 LOAD_CONST               1 (3)
              4 BINARY_RSHIFT
              6 RETURN_VALUE
(venv) kiss@newton add-python % p -m timeit -s 'from shift_vs_div import div8, shift3' 'shift3(478491)'
5000000 loops, best of 5: 41.6 nsec per loop
(venv) kiss@newton add-python % p -m timeit -s 'from shift_vs_div import div8, shift3' 'div8(478491)'  
10000000 loops, best of 5: 38.3 nsec per loop
(venv) kiss@newton add-python % p -m timeit -s 'from shift_vs_div import div8, shift3' 'shift3(478491)'
5000000 loops, best of 5: 41.9 nsec per loop
```

So I'll switch to `// 8`.

Reply:
I'm surprised the compiler even produces different bytecode

Reply:
Turns out there is a subtle difference if the dividend is a float.

```
>>> 10.0 // 8
1.0
>>> 10.0 >> 3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for >>: 'float' and 'int'
```

Reply:
Maybe not so subtle.